### DISCUSSION TITLE
Multiple forcings for one velocity component

### DISCUSSION THREAD

**Discussion by pjrusello** [CliMA/Oceananigans.jl#2134] [Computational science!] [answered]
Is there a recommended way to combine multiple forcings (specifically a `Forcing` and `Relaxation`) for a velocity component? E.g., have both a sponge layer for the top and bottom of the domain but also have a local velocity forcing inside the non-sponge domain?

It seems like I will probably need to re-write my sponge layer function to work with `Forcing` rather than `Relaxation` and combine the sponge layer and velocity forcing in a single function, just curious if there's another/better way to do this.

**Comment by glwagner** [comment, marked as answer]
As you've guessed we don't have an interface for specifying more than one forcing right now, though I think it might be convenient for use cases like yours.

As a temporary workaround before we have such a feature, you can embed a `Relaxation` object _inside_ another forcing function and call it with 

```juia
relaxation(x, y, z, t, f)
```

where `relaxation::Relaxation` and `f` is the field being relaxed (the velocity component in question in this case).

@pjrusello would you like to open an issue with a feature request for specifying multiple forcing functions and we can discuss further there? We might be able to handle this by having users specify forcings as a tuple, eg in a model constructor,

```julia
forcing = (; u = (u_relaxation, u_sponge)),
```

We could consider other API designs too.

**Reply by pjrusello** [reply]
Cool, seems like the `relaxation::Relaxation` route will be pretty straightforward so I'll test that out today. I'll open up an issue for further discussion on this.
